components/brave_ads/core/internal/common/resources/flat/text_processing_neural_model.fbs

```diff
@@ -0,0 +1,51 @@
+namespace text_processing_neural;
+
+table WeightsRow {
+  columns: [float32];
+}
+
+table NeuralMatrixData {
+  weights_rows: [WeightsRow];
+}
+
+table SimpleTransformation {
+  transformation_type: string;
+}
+
+table StringToNumbersMap {
+    code: string (key);
+    numbers: [uint8];
+}
+
+table MappedTokenTransformation {
+  dimension: int32;
+  huffman_coding: [StringToNumbersMap];
+  mapping: [StringToNumbersMap];
+  transformation_type: string;
+}
+
+union Transformation {
+  SimpleTransformation,
+  MappedTokenTransformation
+}
+
+table TransformationEntry {
+  transformation: Transformation;
+}
+
+table Classifier {
+  classes: [string];
+  classifier_type: string;
+  neural_matricies_data: [NeuralMatrixData];
+  neural_post_matrix_functions: [string];
+}
+
+table Model {
+  classifier: Classifier;
+  locale: string;
+  timestamp: string;
+  transformations: [TransformationEntry];
+  version: int;
```

Comment:
Why do we need versioning because when we need to change the version we should download a component with a new id, so that the CRX package is as small as possible.

Reply:
cc @ptjames

Reply:
I suppose we do have validation so that the resource fails to load, so happy to keep version